cmd/tests/events/events.go

```diff
@@ -0,0 +1,74 @@
+// Package events is used for testing the event functionality.
+package events
+
+import (
+	"fmt"
+	"sync"
+
+	"go.k6.io/k6/api/v1/client"
+	"go.k6.io/k6/event"
+	"go.k6.io/k6/js/modules"
+)
+
+// RootModule is the global module instance that will create module
+// instances for each VU.
+type RootModule struct {
+	initOnce        sync.Once
+	apiAddress      string
+	subscribeEvents []event.Type
+}
+
+// Events represents an instance of the events module.
+type Events struct{}
+
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &Events{}
+)
+
+// New returns a pointer to a new RootModule instance.
+func New(apiAddress string, subscribeEvents []event.Type) *RootModule {
+	return &RootModule{
+		initOnce:        sync.Once{},
+		apiAddress:      apiAddress,
+		subscribeEvents: subscribeEvents,
+	}
+}
+
+// NewModuleInstance implements the modules.Module interface to return
+// a new instance for each VU.
+func (rm *RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	rm.initOnce.Do(func() {
+		sid, evtCh := vu.Events().Subscribe(rm.subscribeEvents...)
+		logger := vu.InitEnv().Logger
+		go func() {
+			api, _ := client.New(rm.apiAddress)
+			for {
+				select {
+				case evt, ok := <-evtCh:
+					if !ok {
+						return
+					}
+					var testStatus string
+					if evt.Type != event.Exit {
+						status, _ := api.Status(vu.Context())
+						testStatus = fmt.Sprintf(", test status: %s", status.Status.String())
```

Comment:
I've noticed that this is flaky. :disappointed:

It can happen that the test status has changed between the time the event is received, and this API call returns, so the current test assertions aren't stable.

Any alternative ideas with how to match up k6 execution status with the events we receive here?

Reply:
@olegbespalov It's not just for purposes of logging. This ensures that the event is sent at the right time. I.e. if we receive the `Init` event, that k6 is also in the `InitVUs` stage.

I think this is very valuable, as otherwise the events could be sent at any point, and we have no way of confirming that they represent the actual state in k6. Unfortunately, it's also flaky, and I don't think we have another way of confirming this, other than sending it in the event `Data` as I mentioned above, which wouldn't be ideal.

So if we can't find an alternative, I'll remove it, since we definitely shouldn't have more flaky tests. :sweat_smile:

Reply:
Yeah, you're right here. And since it's a testing module (which I missed), it's not about logging. 

Regarding:
>EDIT: Crazy idea: continuously query the API for the status in the background with somewhat aggressive frequency, and log whatever the last status we received was. I suspect the biggest reason for the flakiness is the time taken to do this network call. But then again, we could always get the event at the "wrong" time and miss the actual test status, so this might be flaky as well...

Still sounds more flaky than your previous idea of just checking the order

Reply:
> Still sounds more flaky than your previous idea of just checking the order

It is. But like I said, the order tells us nothing about _when_ the events were emitted, so we lose an important assertion. But if there's no deterministic alternative, I'll remove it.